### PR TITLE
Feature/672 rage quit

### PR DESCRIPTION
**Pull request by skuhlmann**
## GitHub Issue

fixes https://github.com/HausDAO/daohaus-monorepo/issues/672

## Changes

- ragequit page and route
- ragequit form and tx legos
- custom rq token input (shares or loot)
- custom rg token list select input

## Packages Added

n/a

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [ ] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

File: apps/core-app/src/components/customFields/RagequitToken.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    if (!dao || !connectedMembership) return null;
+    return {
+      label:
+        id === 'sharesToBurn'
+          ? `Voting Tokens (${dao.shareTokenSymbol})`
+          : `Non-Voting Tokens (${dao.lootTokenSymbol})`,
+      maxAmount:
+        id === 'sharesToBurn'
+          ? connectedMembership.shares
+          : connectedMembership.loot,
```
Comment (brossetti1) [inline apps/core-app/src/components/customFields/RagequitToken.tsx:24]: question about loot and shares -- can a member only have 1 or the other, or can they have both?

if both, just curious-- would we want to present that here?

Comment (skuhlmann) [reply apps/core-app/src/components/customFields/RagequitToken.tsx:24]: They can have both - this input is used in both instances - so we have it in the form twice - once for shares and once for loot.

```ts
       fields: [
          {
            id: 'sharesToBurn',
            // @ts-expect-error: cannot resolve type within formSegment using custom fields sent from the core-app
            type: 'ragequitToken',
          },
          // @ts-expect-error: cannot resolve type within formSegment using custom fields sent from the core-app
          { id: 'lootToBurn', type: 'ragequitToken' },
        ],
        ```

File: apps/core-app/src/components/customFields/RagequitTokenList.tsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  };
+
+  const tokenTable = useMemo((): TokenTable | null => {
+    if (!dao || !networkData || !connectedMembership) return null;
+    return dao?.tokenBalances
+      .filter((token) => Number(token.balance) > 0)
+      .reduce(
+        (acc: TokenTable, token: TokenBalance) => {
+          acc.tokenCheckboxes = [
+            ...acc.tokenCheckboxes,
```
Comment (brossetti1) [inline apps/core-app/src/components/customFields/RagequitTokenList.tsx:95]: im going to have to a get a breakdown of how you use reduce -- i usually pick other methods but this is really nice

File: apps/core-app/src/legos/form.ts [hunk 3]
```diff
@@ -543,4 +543,50 @@ export const COMMON_FORMS: Record<string, CustomFormLego> = {
...
+        fields: [
+          {
+            id: 'sharesToBurn',
+            // @ts-expect-error: cannot resolve type within formSegment using custom fields sent from the core-app
+            type: 'ragequitToken',
+          },
+          // @ts-expect-error: cannot resolve type within formSegment using custom fields sent from the core-app
+          { id: 'lootToBurn', type: 'ragequitToken' },
+        ],
+      },
```
Comment (brossetti1) [inline apps/core-app/src/legos/form.ts:564]: so i beleive if you add these fields to https://github.com/HausDAO/daohaus-monorepo/blob/develop/libs/form-builder/src/components/CoreFieldLookup.ts#L19 you do not need the `@ts-expect-error`

**Review by brossetti1 — APPROVED** [review]
added some comments -- approving to not block this but id review the types and ts commetns

File: libs/common-utilities/src/constants/contracts.ts [hunk 4]
```diff
@@ -12,6 +12,9 @@ export type ContractKey =
 export type AddressKeyChain = { [key in ValidNetwork]?: string };
 export type ContractAddressList = Record<ContractKey, AddressKeyChain>;
 
+export const NETWORK_TOKEN_ETH_ADDRESS =
+  '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE';
+
```
[the remark this answers is not in the document]
Comment (skuhlmann) [reply libs/common-utilities/src/constants/contracts.ts:17]: > what does this value do?
> 
> I noticed its used as a fallback for alot of different values

This is used in Baal for the network token address so it's needed in rage quit.

File: apps/core-app/src/legos/form.ts (hunk 3 above)
Comment (skuhlmann) [reply apps/core-app/src/legos/form.ts:564]: right but these are all fields in the form builder itself - not in the core app.

**Pull request merged**